pkg/controllers/nodetemplate/controller.go

```
@@ -96,17 +117,42 @@ func (c *Controller) resolveSubnets(ctx context.Context, nodeTemplate *v1alpha1.
 	return nil
 }
 
-func (c *Controller) resolveSecurityGroup(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate) error {
+func (c *Controller) resolveSecurityGroups(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate) error {
 	securityGroupIds, err := c.securityGroupProvider.List(ctx, nodeTemplate)
 	if err != nil {
 		return err
 	}
+	if len(securityGroupIds) == 0 {
+		nodeTemplate.Status.SecurityGroups = nil
+		logging.FromContext(ctx).Errorf("no securityGroup exist given constraints")
+	}
 
-	nodeTemplate.Status.SecurityGroups = lo.Map(securityGroupIds, func(id string, _ int) v1alpha1.SecurityGroupStatus {
-		return v1alpha1.SecurityGroupStatus{
+	nodeTemplate.Status.SecurityGroups = lo.Map(securityGroupIds, func(id string, _ int) v1alpha1.SecurityGroup {
+		return v1alpha1.SecurityGroup{
 			ID: id,
 		}
 	})
 
 	return nil
 }
+
+func (c *Controller) resolveAMIs(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate) error {
+	amiRequirement, err := c.amiProvider.Get(ctx, nodeTemplate, &amifamily.Options{})
+	if err != nil {
+		return err
+	}
+	if len(amiRequirement) == 0 {
+		nodeTemplate.Status.AMIs = nil
+		logging.FromContext(ctx).Errorf("no amis exist given constraints")
```

We should not just log an error here, I think we want to return an error